### DISCUSSION TITLE
Change size of gaps between windows, but different sizes for left, right, up and bottom edges.

### DISCUSSION THREAD

**Discussion by aravezskinteeth** [rxhanson/RectanglePro-Community#274] [Ideas]
As the title says, I'd like to be able to have different window gap sizes for the left, right, up and bottom. I use sketchybar as a replacement for the menu bar and I need approximately a 45 px gap on the top part so it can fit nicely. However, the gaps on the left, right and bottom parts of the windows are to wide at this particular size. So, I'd love to be able to change the size of the others without affecting the top gap. In other words:

Change these settings:

<img width="855" alt="SCR-20230512-jmwz" src="https://github.com/rxhanson/Rectangle/assets/104392087/40b08816-9c54-42b2-a77c-04deb2f717c2">

So I can stop getting this:

<img width="1680" alt="SCR-20230512-jrjp-2" src="https://github.com/rxhanson/Rectangle/assets/104392087/958585c9-d14a-4e3a-8e56-c22933f9e3e4">

and I get this instead without having to manually resize windows:

<img width="1680" alt="SCR-20230512-jrjp" src="https://github.com/rxhanson/Rectangle/assets/104392087/720e61cf-096d-4b56-99f0-5a83d904fb5d">

**Comment by rxhanson** [comment]
Would you like to just add a screen edge gap? Here is the Teriminal command:

`defaults write com.knollsoft.Hookshot screenEdgeGapTop -int 45`


The others, example:
```
defaults write com.knollsoft.Hookshot screenEdgeGapBottom -int 10
defaults write com.knollsoft.Hookshot screenEdgeGapLeft -int 10
defaults write com.knollsoft.Hookshot screenEdgeGapRight -int 10
```

Then restart the app.

This will be added to the UI in an upcoming release.

If that's not what you're looking for, I would consider adding granular side edge gap configurations if there are enough users that want it.

**Comment by aravezskinteeth** [comment]
It gives me this result, which is too big:

<img width="1680" alt="SCR-20230512-kbxt" src="https://github.com/rxhanson/RectanglePro-Community/assets/104392087/77638cdb-eafc-4475-8795-fede34035ce4">

But after changing the gaps between windows to 10 px on the UI I get exactly what I want, which didn't happened before adding these defaults, so thank you.

<img width="1680" alt="SCR-20230512-kcph" src="https://github.com/rxhanson/RectanglePro-Community/assets/104392087/1a525068-134a-41f6-8581-4769aae98935">

**Comment by eduwass** [comment]
@rxhanson this works! Wondering if the value can be set independently for each monitor?

**Reply by rxhanson** [reply]
There's a `screenEdgeGapTopNotch` value that will be applied instead of `screenEdgeGapTop` only for displays that have the notch, but there is not a per-monitor setting at the moment. This gets asked from time to time, so I'll bump up the priority for this in my backlog.

**Reply by eduwass** [reply]
Hi @rxhanson, that's cool! In my case, I don't have a notch, I'm basically trying to do something similar to what OP is showing, but I just want the top gap in my primary monitor. Thanks for bumping this, appreciate it!
